src/libraries/System.Private.Uri/src/System/UriBuilder.cs

```diff
@@ -255,6 +255,26 @@ public Uri Uri
 
         public override int GetHashCode() => Uri.GetHashCode();
 
+        private static string EncodeUserInfo(string? input)
+        {
+            if (input == null)
+            {
+                return string.Empty;
+            }
+
+            if (input.AsSpan().IndexOfAny(@"/\?#@") < 0)
```

Comment:
```suggestion
            }

            // The following characters ("/" / "\" / "?" / "#" / "@") are from the gen-delims group.
            // We have to escape them to avoid corrupting the rest of the Uri string.
            // Other characters like spaces or non-ASCII will be escaped by Uri, we can ignore them here.
            if (input.AsSpan().IndexOfAny(@"/\?#@") < 0)
```

Just a short comment so we know what this logic is for in the future.

Reply:
That looks good and would be nice to have--I've committed this!